api/backend/blueflood/blueflood.go

```diff
@@ -132,20 +136,32 @@ func (b *blueflood) FetchSingleSeries(request api.FetchSeriesRequest) (api.Times
 	}
 	queryResolution := b.config.bluefloodResolution(
 		request.Timerange.Resolution(),
-		request.Timerange.Start())
+		request.Timerange.Start(),
+	)
 
+	// Sample the data at the given `queryResolution`
 	queryUrl, err := b.constructURL(request, sampler, queryResolution)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
-
-	// Issue GET to fetch metrics
 	parsedResult, err := b.fetch(request, queryUrl)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
 
-	values := processResult(parsedResult, request.Timerange, sampler, queryResolution)
+	combinedResult := parsedResult.Values
+
+	// Sample the data at the FULL resolution.
+	// In order to do this, we use the same `request` object but hard-code the ResolutionFull parameter.
+	fullResolutionQueryURL, err := b.constructURL(request, sampler, ResolutionFull)
+	if err == nil {
+		fullResolutionParsedResult, err := b.fetch(request, fullResolutionQueryURL)
+		if err == nil {
+			combinedResult = append(parsedResult.Values, fullResolutionParsedResult.Values...)
```

Comment:
Should add a comment here about how overlapping data will end up being the same as blueflood's rollups after we do our own. It's not completely intuitive that that's the case.

There may be some situations that this ends up giving wrong data, but I can't think of any. I'm relatively confident this works out though.